### PR TITLE
Search: Migrate more helper classes to package

### PR DESCRIPTION
**Pull request by jsnmoon**
Split off from #21637.

#### Changes proposed in this Pull Request:
* Migrates the template tags helper class, which is primarily responsible for rendering clickable filters within the Jetpack Search widget.
* Migrates the site settings helper class, which is used to persist changes from Customberg onto site options. 

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Apply this change to your site with a Jetpack Search subscription.
2. Add the Jetpack Search widget with configured filters to a site sidebar with only the search module enabled (i.e. instant search disabled).
3. Ensure that the filters from (2) render as expected.
4. Repeat (2) and (3) with Instant Search enabled in addition to the search module.
5. Navigate to Customberg (`/wp-admin/admin.php?page=jetpack-search-configure`).
6. Ensure Customberg reads and saves instant search configuration settings as expected.

### CODE REVIEW

File: projects/packages/search/src/class-template-tags.php [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
- * @package    Jetpack
- * @subpackage Jetpack Search
- * @since      5.8.0
+ * @package    automattic/jetpack-search
  */
 
-use Automattic\Jetpack\Search\Helper;
+namespace Automattic\Jetpack\Search;
+
+use Automattic\Jetpack\Search\Classic_Search as Jetpack_Search;
```
Comment (kangzj) [inline projects/packages/search/src/class-template-tags.php:10]: I couldn't  find `Classic_Search` anywhere, which is in the big migration branch.

**Review by kangzj — CHANGES_REQUESTED** [review]
Hi @jsnmoon it seems `Classic_Search` is missing from the PR.

File: projects/packages/search/src/class-template-tags.php (hunk 1 above)
Comment (jsnmoon) [reply projects/packages/search/src/class-template-tags.php]: Ah, good catch. This change wasn't intended for this PR.

File: projects/plugins/jetpack/modules/widgets/search.php [hunk 2]
```diff
@@ -401,6 +401,7 @@ public function widget_non_instant( $args, $instance ) {
 			 */
 			do_action(
 				'jetpack_search_render_filters',
+				Jetpack::get_option( 'id' ),
```
[the remark this answers is not in the document]
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/widgets/search.php]: This change was accidentally included in this PR.

In the migrated versions of Classic_Search and Instant_Search, `static::instance` must be invoked with the blog ID. I don't think it's desirable, but we can discuss that in the migration PR.

**Comment by jsnmoon** [comment]
> Hi @jsnmoon it seems `Classic_Search` is missing from the PR. (@kangzj)

It was never meant to be included in this PR; fixed! Good catch! 😅

**Comment by kangzj** [comment]
Hi @jsnmoon unfortunately this broke `/wp-admin/widgets.php` for me. I don't see any errors, only a blank screen. Could you have a look at this while I continue my tests and review? Thanks

File: projects/plugins/jetpack/modules/widgets/search.php [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<?php
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/widgets/search.php:1]: Could we bring class `Jetpack_Search_Widget` over btw? only leave the init process in this file.

Comment (jsnmoon) [reply projects/plugins/jetpack/modules/widgets/search.php:1]: The widget will likely migrate during milestone 2 :)

**Comment by jsnmoon** [comment]
> Hi @jsnmoon unfortunately this broke `/wp-admin/widgets.php` for me. I don't see any errors, only a blank screen. Could you have a look at this while I continue my tests and review? (@kangzj)

I noticed this went away after I merged in the latest master, so I'm guessing it was an unrelated regression. It should be fixed now!

> Other than that, the widget works fine for both Classic and Instant Search. Another thing I noticed that, we'd probably need a version bump by running command: `tools/project-version.sh -f -u 0.2.0-alpha packages/search` (@kangzj)

Done!

**Review by kangzj — DISMISSED** [review]
Well done Jason, this looks good and test well 👍

**Review by jeherve — CHANGES_REQUESTED** [review]
This is going to need a rebase I'm afraid. 

You should also be able to remove the now error-free file from `tools/phpcs-excludelist.json`.

**Comment by kangzj** [comment]
Hi @jeherve rebased and removed the file from phpcs excluded list. Thanks.

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php [hunk 4]
```diff
@@ -63,8 +63,7 @@ public static function instance() {
 	public function load_php() {
 		$this->base_load_php();
 
-		require_once __DIR__ . '/class-jetpack-search-settings.php';
-		new Jetpack_Search_Settings();
+		new Automattic\Jetpack\Search\Settings();
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:66]: Is this needed now that the class will be autoloaded?

**Review by jeherve — APPROVED** [review]
This looks good to me. Let's merge this. I had a question, but that's something that can be discussed after the merge I think.

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php (hunk 4 above)
Comment (kangzj) [reply projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:66]: It's needed as the class needs to be initialized, which added some actions/filters. Thanks

**Comment by kangzj** [comment]
These changes will be synced to WPCOM once the classic/instant search migration is complete.

**Pull request merged**